VM Agent e2e test failed because of the change in support bundle content

**Describe the bug**
<!--
A clear and concise description of what the bug is.
If you believe this bug is a security issue, please don't use this template and follow our [security guidelines](/SECURITY.md)
-->
Hi,

VM Agent e2e test failed with the below error,

```
    vmagent_test.go:224: Actual files after extracting SupportBundleCollection tarball vmbmtest0-win-0_support-bundle-collection-external-node: [addressgroups agentinfo appliedtogroups flows goroutinestacks groups ipconfig logs\ovs\ovs-vswitchd.log logs\ovs\ovsdb-server.log memprofile network-adapters networkpolicies ovsports routes]
    vmagent_test.go:225: 
        	Error Trace:	/var/lib/jenkins/workspace/antrea-vm-e2e-for-pull-request/test/e2e/vmagent_test.go:225
        	            				/var/lib/jenkins/workspace/antrea-vm-e2e-for-pull-request/test/e2e/vmagent_test.go:82
        	Error:      	elements differ
        	            	
        	            	extra elements in list B:
        	            	([]interface {}) (len=1) {
        	            	 (string) (len=6) "groups"
        	            	}
        	            	
        	            	
        	            	listA:
        	            	([]string) (len=13) {
        	            	 (string) (len=13) "addressgroups",
        	            	 (string) (len=9) "agentinfo",
        	            	 (string) (len=15) "appliedtogroups",
        	            	 (string) (len=5) "flows",
        	            	 (string) (len=15) "goroutinestacks",
        	            	 (string) (len=8) "ipconfig",
        	            	 (string) (len=25) "logs\\ovs\\ovs-vswitchd.log",
        	            	 (string) (len=25) "logs\\ovs\\ovsdb-server.log",
        	            	 (string) (len=10) "memprofile",
        	            	 (string) (len=16) "network-adapters",
        	            	 (string) (len=15) "networkpolicies",
        	            	 (string) (len=8) "ovsports",
        	            	 (string) (len=6) "routes"
        	            	}
        	            	
        	            	
        	            	listB:
        	            	([]string) (len=14) {
        	            	 (string) (len=13) "addressgroups",
        	            	 (string) (len=9) "agentinfo",
        	            	 (string) (len=15) "appliedtogroups",
        	            	 (string) (len=5) "flows",
        	            	 (string) (len=15) "goroutinestacks",
        	            	 (string) (len=6) "groups",
        	            	 (string) (len=8) "ipconfig",
        	            	 (string) (len=25) "logs\\ovs\\ovs-vswitchd.log",
        	            	 (string) (len=25) "logs\\ovs\\ovsdb-server.log",
        	            	 (string) (len=10) "memprofile",
        	            	 (string) (len=16) "network-adapters",
        	            	 (string) (len=15) "networkpolicies",
        	            	 (string) (len=8) "ovsports",
        	            	 (string) (len=6) "routes"
        	            	}
        	Test:       	TestVMAgent/testExternalNodeSupportBundleCollection
...
    --- FAIL: TestVMAgent/testExternalNodeSupportBundleCollection (15.89s)
```

e2e link: http://10.176.27.169:8080/job/antrea-vm-e2e-for-pull-request/586

The failure is because of this change https://github.com/antrea-io/antrea/pull/6195 which added  OVS groups in the support bundle files, but the VM agent e2e case is not updated accordingly.

**To Reproduce**
<!-- Steps to reproduce the behavior. -->

**Expected**
<!-- A clear and concise description of what you expected to happen. -->

**Actual behavior**
<!-- A clear and concise description of what's the actual behavior. If applicable, add screenshots, log messages, etc. to help explain the problem. -->

**Versions:**
<!--
Please provide the following information:
 - Antrea version (Docker image tag).
 - Kubernetes version (use `kubectl version`). If your Kubernetes components have [different versions](https://kubernetes.io/docs/setup/release/version-skew-policy/), please provide the version for all of them.
 - Container runtime: which runtime are you using (e.g. containerd, cri-o, docker) and which version are you using?
 - Linux kernel version on the Kubernetes Nodes (`uname -r`).
 - If you chose to compile the Open vSwitch kernel module manually instead of using the kernel module built into the Linux kernel, which version of the OVS kernel module are you using? Include the output of `modinfo openvswitch` for the Kubernetes Nodes.
-->

**Additional context**
<!-- Add any other context about the problem here, such as Antrea logs, kubelet logs, etc. -->

<!-- (Please consider pasting long output into a [GitHub gist](https://gist.github.com) or any other pastebin.) -->

--- a/test/e2e/vmagent_test.go
+++ b/test/e2e/vmagent_test.go
@@ -216,13 +216,17 @@ func testExternalNodeSupportBundleCollection(t *testing.T, data *TestData, vmLis
 		require.NoError(t, err)
 		var expectedInfoEntries []string
 		if vm.osType == linuxOS {
-			expectedInfoEntries = []string{"address", "addressgroups", "agentinfo", "appliedtogroups", "flows", "goroutinestacks", "iptables", "link", "logs", "memprofile", "networkpolicies", "ovsports", "route"}
+			expectedInfoEntries = []string{"address", "addressgroups", "agentinfo", "appliedtogroups", "flows", "goroutinestacks", "groups", "iptables", "link", "logs", "memprofile", "networkpolicies", "ovsports", "route"}
 		} else if vm.osType == windowsOS {
-			expectedInfoEntries = []string{"addressgroups", "agentinfo", "appliedtogroups", "flows", "goroutinestacks", "ipconfig", "logs\\ovs\\ovs-vswitchd.log", "logs\\ovs\\ovsdb-server.log", "memprofile", "network-adapters", "networkpolicies", "ovsports", "routes"}
+			expectedInfoEntries = []string{"addressgroups", "agentinfo", "appliedtogroups", "flows", "goroutinestacks", "groups", "ipconfig", "logs\\ovs\\ovs-vswitchd.log", "logs\\ovs\\ovsdb-server.log", "memprofile", "network-adapters", "networkpolicies", "ovsports", "routes"}
 		}
-		actualExpectedInfoEntries := strings.Split(strings.Trim(stdout, "\n"), "\n")
-		t.Logf("Actual files after extracting SupportBundleCollection tarball %s_%s: %v", vm.nodeName, bundleName, actualExpectedInfoEntries)
-		assert.ElementsMatch(t, expectedInfoEntries, actualExpectedInfoEntries)
+		actualInfoEntries := strings.Split(strings.Trim(stdout, "\n"), "\n")
+		t.Logf("Actual files after extracting SupportBundleCollection tarball %s_%s: %v", vm.nodeName, bundleName, actualInfoEntries)
+		// We validate that actualInfoEntries contains expectedInfoEntries instead
+		// of checking for an exact match, which would make the test too easy to break. It
+		// is recommended to update expectedInfoEntries when new elements are added to the
+		// supportbundle, but it is not a strict requirement.
+		assert.Subset(t, actualInfoEntries, expectedInfoEntries)
 	}
 }
 

